### PR TITLE
For #18036 - Add TrackKeyInfo to Ad Click Metrics

### LINKED ISSUE #18036: Add more detail to Ad Click Metrics
In `AdsTelemetry` we currently recognize clicked ads and reports those all as `SearchAdClicked` event that just have the `providerName`.

https://github.com/mozilla-mobile/fenix/blob/07748f69b75144e8e247b742f66a756f999cb299/app/src/main/java/org/mozilla/fenix/search/telemetry/ads/AdsTelemetry.kt#L37

We need to add more detail to these metrics, in a similar way that is done as in `InContentTelemetry`:

https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/search/telemetry/incontent/InContentTelemetry.kt#L51

This should also include the `channel` that we introduced for Top Site Tile work.

The exact format of the event is TBD and needs to be worked out with Data Science.

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #18036 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Review by codrut-topliceanu — APPROVED** [review]
Ran a few local tests, and it seems in order.
Looks good to me. 👍  But I hope @st3fan can give a second opinion just in case.

**Comment by gabrielluong** [comment]
sorry @codrut-topliceanu, can I have you re-review and test again? Thanks!

File: app/src/main/assets/extensions/ads/adsBackground.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (gabrielluong) [inline app/src/main/assets/extensions/ads/adsBackground.js:1]: This is a copy of https://github.com/mozilla-mobile/fenix/blob/07748f69b75144e8e247b742f66a756f999cb299/app/src/main/assets/extensions/cookies/cookiesBackground.js.

File: app/src/main/assets/extensions/ads/ads.js [hunk 2]
```diff
@@ -14,28 +14,39 @@ function collectLinks(urls) {
...
+browser.runtime.onMessage.addListener(notify);
+
 const events = ["pageshow", "load", "unload"];
 var timeout;
 
 const eventLogger = event => {
   switch (event.type) {
     case "load":
-        timeout = setTimeout(sendLinks, ADLINK_CHECK_TIMEOUT_MS);
+        timeout = setTimeout(() => {
```
Comment (gabrielluong) [inline app/src/main/assets/extensions/ads/ads.js:41]: Most of these changes are copied from https://github.com/mozilla-mobile/fenix/blob/07748f69b75144e8e247b742f66a756f999cb299/app/src/main/assets/extensions/cookies/cookies.js in order to retrieve the cookie information and send the cookies JSON message.

File: app/src/main/assets/extensions/ads/manifest.template.json [hunk 3]
```diff
@@ -19,9 +19,20 @@
...
     }
   ],
+  "background": {
+    "scripts": ["adsBackground.js"]
+  },
   "permissions": [
     "geckoViewAddons",
     "nativeMessaging",
-    "nativeMessagingFromContent"
+    "nativeMessagingFromContent",
```
Comment (gabrielluong) [inline app/src/main/assets/extensions/ads/manifest.template.json:28]: Permissions are also reused from the cookies https://github.com/mozilla-mobile/fenix/blob/07748f69b75144e8e247b742f66a756f999cb299/app/src/main/assets/extensions/cookies/manifest.template.json.

File: app/src/main/java/org/mozilla/fenix/search/telemetry/SearchProviderModel.kt [hunk 4]
```diff
@@ -39,7 +39,7 @@ data class SearchProviderModel(
      * Checks if any of the given URLs represent an ad from the search engine.
      * Used to check if a clicked link was for an ad.
      */
-    fun containsAds(urlList: List<String>) = urlList.any { url -> isAd(url) }
+    fun hasAdClicks(urlList: List<String>) = urlList.any { url -> isAd(url) }
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/search/telemetry/SearchProviderModel.kt:42]: This was a simple refactor of the naming. I think this is closer to our understanding of the search telemetry terminologies.

Comment (Standard8) [reply app/src/main/java/org/mozilla/fenix/search/telemetry/SearchProviderModel.kt:42]: I find the name a little strange - you're checking a url list for any of the urls being ad links, but then you're calling it a "click".

The function appears to be used for both when a SERP page might contain ad links and when a user clicks a link.

I think `containsAds` was probably more appropriate, or maybe even better would be `containsAdLinks` seeing as you're testing a list for ad links.

**Review by codrut-topliceanu — APPROVED** [review]
LGTM! 👍

**Pull request merged**